assets/js/components/DateRangeSelector.js

```diff
@@ -40,32 +45,11 @@ function DateRangeSelector() {
 	const { setDateRange } = useDispatch( CORE_USER );
 
 	const [ menuOpen, setMenuOpen ] = useState( false );
-	const menuButtonRef = useRef();
-	const menuRef = useRef();
-
-	useEffect( () => {
-		const handleMenuClose = ( event ) => {
-			if ( ( menuButtonRef && menuButtonRef.current ) && ( menuRef && menuRef.current ) ) {
-				// Close the menu if the user presses the Escape key
-				// or if they click outside of the menu.
-				if (
-					( ( 'keyup' === event.type && ESCAPE === event.keyCode ) || 'mouseup' === event.type ) &&
-					! menuButtonRef.current.contains( event.target ) &&
-					! menuRef.current.contains( event.target )
-				) {
```

Comment:
(as mentioned on the ticket) 

This looks like a bug and there should be extra parenthesis after the ||

This means that if you open the menu pressing enter, then close pressing escape, you have to hit enter twice to open it again (same bug as above with the first keypress resetting the state)